apisix/plugins/file-logger.lua

```diff
@@ -70,10 +70,15 @@ local function write_file_data(conf, log_message)
     local file, err = io_open(conf.path, 'a+')
 
     if not file then
-        core.log.error("failed to open file: " .. conf.path .. ", error info: " .. err)
+        core.log.error("failed to open file: ", conf.path, ", error info: ", err)
     else
-        file:write(msg)
-        file:flush()
+        local ok, error = file:write(msg)
```

Comment:
```suggestion
        local ok, err = file:write(msg)
```
and we need to close the file even failed to write.

BTW, we can use `file:write(msg, '\n')` to avoid `local msg = core.json.encode(log_message) .. "\n"`